x/protocolpool/keeper/keeper.go

```diff
@@ -90,6 +102,126 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while iterating all the continuous funds: %w", err)
+	}
+	// get allocated continuous fund
+	fundsAllocated, err := k.getAndResetDistributedFunds(ctx, recipient)
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while getting distributed funds: %w", err)
+	}
+
+	cf, err := k.ContinuousFund.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no continuous fund found for recipient: %s", recipient.String())
+		}
+		return sdk.Coin{}, err
+	}
+
+	recipientBal := k.bankKeeper.GetAllBalances(ctx, recipient)
+	recipientAmount := recipientBal.AmountOf(denom)
+	totalRecipientBal := recipientAmount.Add(fundsAllocated)
+	// check if the recipient account balance exceeds maxDistributedCapital after distribution
+	var withdrawnAmount sdk.Coin
+	if totalRecipientBal.LT(cf.MaxDistributedCapital) {
+		withdrawnAmount = sdk.NewCoin(denom, fundsAllocated)
+		// Distribute funds to the recipient from pool module account
+		err := k.DistributeFromCommunityPool(ctx, sdk.NewCoins(withdrawnAmount), recipient)
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+
+		// decrement fundsAllocated from ToDistribute
+		toDistribute, err := k.ToDistribute.Get(ctx)
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+		err = k.ToDistribute.Set(ctx, toDistribute.Sub(fundsAllocated))
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+	}
+
+	return withdrawnAmount, nil
+}
+
+func (k Keeper) iterateAndUpdateFundsDistribution(ctx context.Context) (string, error) {
+	totalPercentageToBeDistributed := math.ZeroInt()
+	err := k.RecipientFundPercentage.Walk(ctx, nil, func(key sdk.AccAddress, value math.Int) (stop bool, err error) {
+		totalPercentageToBeDistributed = totalPercentageToBeDistributed.Add(value)
+		return false, nil
+	})
+	if err != nil {
+		return "", err
+	}
+	if totalPercentageToBeDistributed.GT(math.NewInt(100)) {
+		return "", fmt.Errorf("total funds percentage cannot exceed 100")
+	}
+
+	poolMAcc := k.authKeeper.GetModuleAccount(ctx, types.ModuleName)
+
+	distrMAcc := k.authKeeper.GetModuleAccount(ctx, distrtypes.ModuleName)
+	distrBal := k.bankKeeper.GetAllBalances(ctx, distrMAcc.GetAddress())
+	distrDecAmount := sdk.NewDecCoinsFromCoins(distrBal...)
+
+	var denom string
+	err = k.RecipientFundPercentage.Walk(ctx, nil, func(key sdk.AccAddress, value math.Int) (stop bool, err error) {
+		// Calculate the funds to be distributed based on the percentage
+		distributionAmount := distrDecAmount.MulDec(math.LegacyNewDecFromIntWithPrec(value, 2))
+		denom = distributionAmount.GetDenomByIndex(0)
+		distrAmount := distributionAmount.AmountOf(denom)
+		distrCoins := distrAmount.TruncateInt()
+
+		// Send distribution funds to pool module account [i.e., poolMAcc = MAcc + toDistribute]
+		err = k.bankKeeper.SendCoinsFromModuleToModule(ctx, distrMAcc.GetName(), poolMAcc.GetName(), sdk.NewCoins(sdk.NewCoin(denom, distrCoins)))
```

Comment:
i dont think this is correct. in a previous pr we pushed the funds to the protocol pool, now we are sending the funds from distr account to protocol pool. I dont think its safe for protocol pool to move funds from another module. This is something we should prevent in the future, this module should not be allowed to move another modules funds.

Reply:
+1. x/distribution sends the community pool tax to x/protocolpool. I don't think a continuous fund pool proposal should directly take fund from distribution (so inflation) (in addition to the community pool tax), but a percentage from the community pool. That way no need to take funds from distribution.